doc/install/install-openshift.md

```diff
@@ -0,0 +1,214 @@
+export REDIS_URL=redis://user:xxxxxx@some-redis.com:10340
+
+```
+
+## Installing the dependencies
+
+Kuadrant integrates with Istio as a Gateway API provider. Before trying Kuadrant, we need to setup an Istio based Gateway API provider. For this we will use the `Sail` operator. 
+
+- Install v1 of Gateway API
+
+```
+kubectl apply -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.0.0/standard-install.yaml
+```
+
+- ### Install and Configure Istio via the Sail Operator
+
+
+#### Install
+```
+kubectl create ns istio-system
+```
+
+```
+kubectl  apply -f - <<EOF
+kind: OperatorGroup
+apiVersion: operators.coreos.com/v1
+metadata:
+  annotations:
+    argocd.argoproj.io/sync-wave: "0"
+  name: sail
+  namespace: istio-system
+spec: 
+  upgradeStrategy: Default  
+---  
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: sailoperator
+  namespace: istio-system
+spec:
+  channel: 3.0-dp1
+  installPlanApproval: Automatic
+  name: sailoperator
+  source: community-operators
+  sourceNamespace: openshift-marketplace
+EOF
+```
+
+To check the status of the install you can run:
+
+```
+kubectl get installplan -n istio-system -o=jsonpath='{.items[0].status.phase}'
+```
+
+once ready it will be marked `complete` while installing it will be marked `installing`
+
+#### Configure Istio
+
+```
+kubectl apply -f - <<EOF
+apiVersion: operator.istio.io/v1alpha1
+kind: Istio
+metadata:
+  annotations:
+    argocd.argoproj.io/sync-wave: "1"
+  name: default
+spec:
+  version: v1.21.0
+  namespace: istio-system
+  # Disable autoscaling to reduce dev resources
+  values:
+    pilot:
+      autoscaleEnabled: false
+EOF
+```
+
+Wait for Istio to be ready
+
+```
+kubectl wait istio/default -n istio-system --for="condition=Ready=true"
+```
+
+
+### (Recommended) Thanos and Observability Stack
+
+Kuadrant provides a set of sample dashboards that use the known metrics exported by kuadrant and gateway components to provide insight into the different areas of your APIs and Gateways. While this is not essential it is recommended that you set up an observability stack. Below are links to the OpenShift docs on this and also a link to help with the setup of Thanos for metrics storage.
+
+OpenShift supports a user facing monitoring stack. This can be cofigured and setup via their documentation
+https://docs.openshift.com/container-platform/4.14/observability/monitoring/configuring-the-monitoring-stack.html
+
+- If you have user workload monitoring enabled. We Recommend configuring  remote write to a central storage system such as Thanos: 
+- [Remote Write Config](https://docs.openshift.com/container-platform/4.14/observability/monitoring/configuring-the-monitoring-stack.html#configuring_remote_write_storage_configuring-the-monitoring-stack)
+
+- [Kube Thanos](https://github.com/thanos-io/kube-thanos)
+
+- [Kuadrant Examples](https://docs.kuadrant.io/kuadrant-operator/doc/observability/examples/)
```

Comment:
I'm thinking about enhancing the setup instructions for Thanos and the Observability Stack to help users get up and running more quickly. I agree that they can find everything important in the guides provided, but the idea is to help them get whole stuck up and running quickly.

Reply:
Opening up a new repo is great, but this par only points to 4 other links, and its informative value is low. I am currently going through the documentation mentioned above and trying to get it to work.
We are documenting how to install Sail, so it does make sense to do the same for the Observability stack.

Reply:
No the observability stack is optional and far more complex to maintain. Istio as a gateway provider is a hard dependency

Reply:
I would argue that everything we are showing is optional, but observability was one of the requirements for this release and a big feature. If the scope of this doc is to showcase everything important we should make more effort to help users get up to speed and to help them set up the feature. We do not need to copy paste openshift doc but we could link to specific sections that might be of particular interest for the simplest use case we are trying to show.

Also Grafana is not covered as part of openshift guide and should have some instruction how to install as well

Reply:
> do we have a good ref on grafana install

maybe https://cloud.redhat.com/experts/o11y/ocp-grafana/ ?
It can be installed from operatorhub as well I believe

> enhancing the setup instructions for Thanos and the Observability Stack

To keep things easier for this version, and also because of known issues with aggregations from multiple clusters in dashboards, we could direct users to configure UWM in 1 cluster only (no remote write instructions). 
Then deploy anything needed to get UWM scraping from istio, envoy, gateway-api-state-metrics (via custom install of kube-state-metrics).
Finally, point to the new kuadrant dashboards doc for how to import dashboards.

I can help with pointing to the right resources for this if agreeable?

Reply:
@david-martin do you want to add PR to this branch with any changes you feel make sense